sound/soc/intel/boards/sof_sdw_dmic.c

```diff
@@ -5,21 +5,6 @@
  *  sof_sdw_dmic - Helpers to handle dmic from generic machine driver
  */
 
-#include <linux/acpi.h>
-#include <linux/async.h>
-#include <linux/delay.h>
-#include <linux/device.h>
-#include <linux/gpio.h>
-#include <linux/init.h>
-#include <linux/input.h>
-#include <linux/io.h>
-#include <linux/dmi.h>
-#include <linux/module.h>
-#include <linux/slab.h>
-#include <sound/jack.h>
-#include <sound/pcm.h>
-#include <sound/pcm_params.h>
-#include <sound/soc.h>
```

Comment:
@plbossart please, let's not do this. Here's an excerpt from Documentation/process/submit-checklist.rst:

> 1) If you use a facility then #include the file that defines/declares
   that facility.  Don't depend on other header files pulling in ones
   that you use.

sof_sdw_common.h really shouldn't be an "include it all" header.

Reply:
Thanks for volunteering to identify which include files are needed for each .c file :-)

Reply:
Sure, I can gladly help with that. But now we have a working version, I don't see a reason to make it worse. If someone finds a problem - superfluous or missing headers - that can always be fixed. It's difficult to have it perfect and I don't think it's expected. I'm pretty sure many or even most files in the kernel have less than perfect header sets. It's just good to make a reasonable effort to have some good approximation of that perfect header set. If someone finds an issue - it can always be fixed. But this kind of change really makes it worse and contradicts the kernel style requirements.